### PR TITLE
Fix bug frequency issue alan-turing-institute#866 with test file

### PR DESCRIPTION
**Pull request by xiaobenbenecho**
- Fix issue ExponentialSmoothing results in error for a series with w-sun freq #866, in _get_fre in utils.datetime to extract the variant frequency of the series, e.g. 'w-sun'.
- Add associated test file.

### CODE REVIEW

**Comment by fkiraly** [comment]
If you want to check which tests fail remotely (and which linter, for instance), click on "details" here:
image
then on "errors/warnings" here:
image
then here:
image
then here:
image

File: sktime/utils/datetime.py
Comment (fkiraly) [inline sktime/utils/datetime.py:4]: This should say something descriptive - e.g., `pandas.datetime related utilities`

File: sktime/utils/tests/test_datetime.py
Comment (fkiraly) [inline sktime/utils/tests/test_datetime.py:10]: this is not a fixture, is it? this is a test

Comment (fkiraly) [reply sktime/utils/tests/test_datetime.py:10]: so I think `@pytest.fixture` should be removed

**Comment by fkiraly** [comment]
Still a doc quality check failing 😃 don't worry, first time round it's this fiddly for everyone.

Join us on Friday, and I (or a colleague) can show you how to set up the code/doc checks locally.

**Comment by xiaobenbenecho** [comment]
The good news for commit be437bf
<https://github.com/alan-turing-institute/sktime/pull/1409/commits/be437bf6b1fa39a96758bc425136e0c06ca8bbed>
is
probably only the doc-quality check failed...
Yeah, will join the meet-up on Friday. thanks.

Franz Király ***@***.***> 于2021年9月13日周一 下午6:07写道：

**Comment by xiaobenbenecho** [comment]
Screenshot 2021-09-13 at 21 53 57
Got this error after pull the newest updates from the main and merge into the branch. Not sure how to resolve this.

**Comment by fkiraly** [comment]
... what is this??

never seen this before. Did azure pipelines change their T&C?

FYI, @mloning, @TonyBagnall

**Comment by fkiraly** [comment]
anyway, the issue with code quality is unrelated to this and an actual linting issue

**Comment by fkiraly** [comment]
Re. parallelism issue, see #1415, core devs are aware - this is Microsoft switching off free ADO pipelines (for unknown reason).

**Comment by fkiraly** [comment]
So, still some doc/code issue, @xiaobenbenecho - looking forward seeing you in tomorrow's community collaborative session (discord, 1pm UTC) if you would like some tipps on this.

**Comment by xiaobenbenecho** [comment]
> So, still some doc/code issue, @xiaobenbenecho - looking forward seeing you in tomorrow's community collaborative session (discord, 1pm UTC) if you would like some tipps on this. 😄

Yeah, will be there. see you tomorrow.

**Comment by xiaobenbenecho** [comment]
> > So, still some doc/code issue, @xiaobenbenecho - looking forward seeing you in tomorrow's community collaborative session (discord, 1pm UTC) if you would like some tipps on this. 😄
> 
> Yeah, will be there. see you tomorrow.

I think I am in the right session. Seems no one shows up at the moment...Let me know if I entered the wrong channel.

**Comment by fkiraly** [comment]
>Seems no one shows up at the moment...Let me know if I entered the wrong channel.

I think you are in the right session - I just joined, but some people only join for one of the two hours.

**Comment by fkiraly** [comment]
you can also get in touch via slack (there is always someone there) or gitter, it just seems there was low attendance in the first hour today. We're there now, @xiaobenbenecho, in case you're still available.

**Comment by xiaobenbenecho** [comment]
Seems all checks have been passed.:)

Franz Király ***@***.***> 于2021年9月17日周五 下午3:08写道：

File: sktime/utils/datetime.py
Comment (fkiraly) [inline sktime/utils/datetime.py:6]: the file docstring should describe the contents of the file, not the change you made (that goes in the commit message)

File: sktime/utils/datetime.py
Comment (fkiraly) [inline sktime/utils/datetime.py:9]: you should add yourself to the authors (GitHub ID) and change markus to `"mloning"`

File: sktime/utils/datetime.py
Comment (fkiraly) [inline sktime/utils/datetime.py:21]: the description for `freq` is incomplete, can you add this? I know it's not you who wrote it, but while we're working on this.

**Review by fkiraly — CHANGES_REQUESTED** [review]
This is nice, thanks for the fix!

I have only minor comments about docstrings inside.

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:6]: Sure. Will change it.

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:9]: Thanks, will add my name to the author list. (just in alignment with the author list of other files as well)

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:21]: Happy to. Will complete the description of 'freq'.

**Comment by fkiraly** [comment]
FYI, if you think it's ready for another review, you can indicate this by clicking the arrows at the top here:
image
(then an automated message is sent)

**Comment by xiaobenbenecho** [comment]
> FYI, if you think it's ready for another review, you can indicate this by clicking the arrows at the top here:
> image
> (then an automated message is sent)

Haha, lots of GitHub functions I never heard about before. Really handy though.

File: sktime/utils/datetime.py
Comment (fkiraly) [inline sktime/utils/datetime.py:4]: this should probably be "date/time format related utilities" or similar?

File: sktime/utils/datetime.py
Comment (fkiraly) [inline sktime/utils/datetime.py:7]: this should be GitHub IDs, i.e., `mloning` etc

File: sktime/utils/datetime.py
Comment (fkiraly) [reply sktime/utils/datetime.py:19]: should include what it can be - I think it should be `valid pandas.datetime freq string`?

Comment (fkiraly) [reply sktime/utils/datetime.py:19]: (I think `week` etc are not valid strings here)

File: sktime/utils/datetime.py
Comment (fkiraly) [inline sktime/utils/datetime.py:102]: this is not a complete statemen. Suggest `Compute duration to/between time indices.`

**Review by fkiraly — CHANGES_REQUESTED** [review]
Almost ready!
All my comments relate to docstrings and are minor - let's fix them while we are at it.

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:7]: I have added the name to align with the author name format in other files (the original authors seem to put their names instead of the Github IDs). I assume both the author names and GitHub IDs should be OK? Let me know if not, I can surely change it to the Github IDs.

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:4]: I will change it to 'Time format related utilities'?

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:102]: Will make the change according to the suggestion.

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:19]: I see. how about "Frequency of the above duration type"?

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:5]: > Yes let's use GitHub IDs here, that will be more useful, we previously used the author names, but using the GitHub IDs will make it easier to link to the user profiles in the [estimator overview](https://www.sktime.org/en/stable/estimator_overview.html)

OK. I see. Will change to Github IDs then.

File: sktime/utils/datetime.py
Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:66]: > Note that `freqstr` has also become deprecated now, not sure if changing this goes beyond the scope of the current PR.

line 72: if 'freqstr' doesn't exist, then it will return None.
Also, I assume the setup.py file will specify the pandas version. so I assume if Sktime is installed with specified pandas version, hopefully, it should be OK for the user to use.

Comment (xiaobenbenecho) [reply sktime/utils/datetime.py:66]: Let me know whether this makes sense.

**Review by fkiraly — APPROVED** [review]
I'd be happy with this - @mloning, can you comment on whether your `freqstr` related issue is addressed before I merge?

File: sktime/utils/datetime.py
Comment (fkiraly) [reply sktime/utils/datetime.py:66]: I'd say it's beyond scope, but we should open an issue.
What would the "fix" for use of `freqstr` be?

**Comment by fkiraly** [comment]
merged now - thanks, @xiaobenbenecho!

**Comment by xiaobenbenecho** [comment]
> merged now - thanks, @xiaobenbenecho!

Thanks, Franz. This can not be done by your continuous instructions and patient. Hope my next PR would not be this fiddly. :P

**Comment by fkiraly** [comment]
don't worry, this is normal - first PR is always like this!

**Pull request merged**
